src/main/java/com/aws/greengrass/shadowmanager/sync/strategy/SyncStrategy.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.shadowmanager.sync.strategy;
+
+import com.aws.greengrass.shadowmanager.sync.model.SyncContext;
+import com.aws.greengrass.shadowmanager.sync.model.SyncRequest;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Interface to for different functionalities in sync strategies.
+ */
+public interface SyncStrategy {
+    /**
+     * Indicates whether syncing is running or not.
+     */
+    AtomicBoolean syncing = new AtomicBoolean(false);
+
+    /**
+     * Starts syncing the shadows based on the strategy.
+     *
+     * @param context         an context object for syncing
+     * @param syncParallelism number of threads to use for syncing
+     */
+    void startSync(SyncContext context, int syncParallelism);
```

Comment:
minor nit: you could just call this method `start` and the other `stop` - sync is redundant here